### PR TITLE
Ai services refactor

### PR DESCRIPTION
**Pull request by Tommer-Raz**

### CODE REVIEW

File: 01-python/01-python.md [hunk 1]
```diff
@@ -8,25 +7,26 @@ as adequate they may be, there is a long and arduous journey yet ahead of you,
...
     4. [google pyguide](https://google.github.io/styleguide/pyguide.html)
     5. [OOP and python](https://realpython.com/inheritance-composition-python/#whats-inheritance)
-2. ###### regex
+
+2. #### Regex
     1. https://www.w3schools.com/python/python_regex.asp
     2. Create regex for Email address (use [this site](https://regex101.com))
-3. ###### packages
-    3. [Python package](https://packaging.python.org/tutorials/packaging-projects/)
-    4. [Build cli - click](https://pymbook.readthedocs.io/en/latest/click.html)
```
Comment (giladd123) [inline 01-python/01-python.md:22]: Might be useful to be familiar with, we still use click

File: 01-python/01-python.md [hunk 2]
```diff
@@ -38,12 +38,11 @@ for the fruits of the language are manyfold, and in your hands lay but a taste.
...
--   unit test vs integration test
--   fixtures
--   tox
+
     
 ### Final Exercise:
 - Flutter over [w3schools' Python Tutorial + File Handling](https://www.w3schools.com/python/default.asp) and make sure you are familiar with all concepts
 - Do the w3schools [python quiz](https://www.w3schools.com/python/python_quiz.asp)
 - Let your instructor choose for you a random 10 exercises from [w3resource](https://www.w3resource.com/python-exercises/)
+- Write a simple Fastapi server. Use gitlab to manage the code.
```
Comment (giladd123) [inline 01-python/01-python.md:47]: In the environment or www? Also add basic requirements such as access an api have 3 endpoints use different http methods etc

File: 02-docker/02-docker.md [hunk 3]
```diff
@@ -1,5 +1,4 @@
 # Docker
-##### ETC: 4 days
 By the end of this section you will become a docker pro, know what they are,
```
Comment (giladd123) [inline 02-docker/02-docker.md:2]: > know what they are  

This sentence is a wreck

File: 02-docker/02-docker.md [hunk 4]
```diff
@@ -16,6 +15,7 @@ In the final exercise you will gain practical experience.
     3. Don't do stage 3
 4. Read about Configure networking, Manage application data in https://docs.docker.com/network/
 5. https://hamk-sysadmin-docker.github.io/exercises/ - Do everything up to 2.4
+
```
Comment (giladd123) [inline 02-docker/02-docker.md:18]: Just for consistency I think you should add a name to these links

File: 02-docker/02-docker.md [hunk 5]
```diff
@@ -29,5 +29,5 @@ In the final exercise you will gain practical experience.
 -   docker alternatives
 
 ### Final Exercise:
-Create a Dockerfile that packs your code from the previous sections.<br>
-Give an example for a use case where VM is more suitable than a container.
\ No newline at end of file
+- Create a Dockerfile that packs your code of the Fastapi from the previous section.
```
Comment (giladd123) [inline 02-docker/02-docker.md:32]: That packs your FastAPI code*

File: 03-kubernetes&openshift/03.2-openshift.md [hunk 6]
```diff
@@ -1,5 +1,4 @@
 # Openshift
-##### ETC: 3 days
 
 ### TODO:
 Go to https://training-lms.redhat.com/lmt/lmtLogin.prEnc?site=rhls
```
Comment (giladd123) [inline 03-kubernetes&openshift/03.2-openshift.md:4]: Same nitpick about hyperlinks

File: 03-kubernetes&openshift/03.3-helm.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# Helm
```
Comment (giladd123) [inline 03-kubernetes&openshift/03.3-helm.md:1]: Might be more useful to make it about both Helm and Kustomize as they go hand in hand, and Kustomize is difficult to understand on the fly

File: 03-kubernetes&openshift/03.3-helm.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+1. Read about [helm](https://letmegooglethat.com/?q=helm) 
+
+
+### Key Concepts:
+- chart
+- release
+- repo 
+- jinja2
+- templates
```
Comment (giladd123) [inline 03-kubernetes&openshift/03.3-helm.md:13]: Add the concept of a subchart, template functions and kustomization file. Also helm uses go templates, not jinja2

Capital letters might be nice

File: 04-gitlab ci/04-gitlabci.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+- configuration file
+- runner
+- job
+- stage
+- pipeline
+- stage
+- pipeline
+- artifacts
+- environments
+- dependecies
```
Comment (giladd123) [inline 04-gitlab ci/04-gitlabci.md:17]: Stage appears twice, also capital letters

File: 04-gitlab ci/04-gitlabci.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+### Final Exercise:
+Write a CICD pipeline using gitlab ci that:
```
Comment (giladd123) [inline 04-gitlab ci/04-gitlabci.md:20]: Gitlab CI*

File: 04-gitlab ci/04-gitlabci.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+  - Builds the image from the docker file you wrote. (*clue*: Use a build config).
+  - Push the image to quay
+  - Deploy the application to opwnshift.
```
Comment (giladd123) [inline 04-gitlab ci/04-gitlabci.md:23]: opwnshift -> Openshift lol

File: 04-gitlab ci/04-gitlabci.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+- stage
+- pipeline
+- artifacts
+- environments
+- dependecies
+
+### Final Exercise:
+Write a CICD pipeline using gitlab ci that:
+  - Builds the image from the docker file you wrote. (*clue*: Use a build config).
+  - Push the image to quay
```
Comment (giladd123) [inline 04-gitlab ci/04-gitlabci.md:22]: Capital letters

File: 05-jupyter/05-jupyter.md [hunk 13]
```diff
@@ -1,5 +1,4 @@
 # Jupyter
-##### ETC: 1 day
 
 Shall you aim for the moon but you were too drank, jupyter has enough (also enough resources, and you know, that sweet ml blockchain GPUs..)<br>
```
Comment (giladd123) [inline 05-jupyter/05-jupyter.md:3]: Why html in markdown

File: 05-jupyter/05-jupyter.md [hunk 14]
```diff
@@ -24,3 +23,4 @@ Shall you aim for the moon but you were too drank, jupyter has enough (also enou
```
Comment (giladd123) [inline 05-jupyter/05-jupyter.md:23]: I think I had some changes to this chapter on Doobies branch

File: 06-ML basics/06-ml.md [hunk 15]
```diff
@@ -16,21 +15,10 @@ read more about this subject.
...
-    1.  [decision tree](https://www.youtube.com/watch?v=dUetL87cOs8&list=PL7APM8kdz5hNs4R00bWzfRWe5bVnMJjw8&index=8)
-    2.  [ensemble](https://www.youtube.com/watch?v=DiVZ4TNGDjg&list=PL7APM8kdz5hNs4R00bWzfRWe5bVnMJjw8&index=9)
-    3.  [model evaluation](https://www.youtube.com/watch?v=fp_fDfC_jcM&list=PL7APM8kdz5hNs4R00bWzfRWe5bVnMJjw8&index=4)
-
-    **is extremely recommended to watch the entire course, but it is out of this document's scope.**
-
-6. Advanced Topics:
-	1. [Bayesian Optimization] https://machinelearningmastery.com/what-is-bayesian-optimization/
-	2. [Meta Learning](https://www.unite.ai/what-is-meta-learning/)
-	3. [Reinforcement Learning episodes 1-14](https://www.youtube.com/playlist?list=PLZbbT5o_s2xoWNVdDudn51XM8lOuZ_Njv)
```
Comment (giladd123) [inline 06-ML basics/06-ml.md:33]: Reinforcement learning is kinda nice

File: 06-ML basics/06-ml.md [hunk 16]
```diff
@@ -55,7 +43,4 @@ read more about this subject.
 
 ### Final Exercise:
 Create a model that predicts avocado prices. You can find the dataset in this folder.
```
Comment (giladd123) [inline 06-ML basics/06-ml.md:45]: Create in env, use runai and mlflow, the dataset is already whitened

File: 08-RunAI/08-RunAI.md [hunk 17]
```diff
@@ -15,4 +15,8 @@
 -   over-quotas
 
 ### Final Exercise:
-Oral exam
+- Oral exam
+- Train your model from ML basic in a training job. Upload your model weight files to a S3 bucket.
```
Comment (giladd123) [inline 08-RunAI/08-RunAI.md:19]: an S3 bucket*

File: 10-AI Services/10-ai services.md [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+
+### TODO:
+
+- [What are ai services](https://www.techtarget.com/searchenterpriseai/definition/Artificial-Intelligence-as-a-Service-AIaaS)
+- Read about AI services on different cloud providers (AWS, GCP, ...)
+
+### Key Concepts:
+- AI Services
+- Types of services and usecases
+- huggingface
```
Comment (giladd123) [inline 10-AI Services/10-ai services.md:11]: Capital

File: 10-AI Services/10-ai services.md [hunk 19]
```diff
@@ -0,0 +1,16 @@
+# AI Services
+
+### TODO:
+
+- [What are ai services](https://www.techtarget.com/searchenterpriseai/definition/Artificial-Intelligence-as-a-Service-AIaaS)
```
Comment (giladd123) [inline 10-AI Services/10-ai services.md:5]: AI Services*

File: 10-AI Services/10-ai services.md [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+
+### Key Concepts:
+- AI Services
+- Types of services and usecases
+- huggingface
+
+### Final Exercise:
+- Oral exam
+- Find a model from huggingface (Should be a small model) and deploy it using huggingface's transformers framework.
+- Ask one of your team mates (not your trainer), what types of ai services we provide.
```
Comment (giladd123) [inline 10-AI Services/10-ai services.md:16]: > team mates  

teammates

> ai services

AI Services

File: 10-AI Services/10.1-LLMs.md [hunk 21]
```diff
@@ -0,0 +1,16 @@
+# Gitlab CI
```
Comment (giladd123) [inline 10-AI Services/10.1-LLMs.md:1]: Huh?

File: 10-AI Services/10.1-LLMs.md [hunk 22]
```diff
@@ -0,0 +1,16 @@
...
+
+### TODO:
```
Comment (giladd123) [inline 10-AI Services/10.1-LLMs.md:3]: Add "what are llms"

File: 10-AI Services/10.1-LLMs.md [hunk 23]
```diff
@@ -0,0 +1,16 @@
...
+- [How LLMs work](https://medium.com/data-science-at-microsoft/how-large-language-models-work-91c362f5b78f)
+- [Evolution of LLMs](https://medium.com/@simon_attard/the-evolution-of-llms-over-the-last-12-months-188a04edb3ac)
+
+### Key Concepts:
+- state of the art LLMs
+- tokens
+- prompts
+- temperature
+- finetuning
+- common LLM parameters
```
Comment (giladd123) [inline 10-AI Services/10.1-LLMs.md:13]: CAPITAL LETTERS

File: 10-AI Services/10.2-bentoml.md [hunk 24]
```diff
@@ -0,0 +1,19 @@
...
+### TODO:
+
+- [Bentoml vs Other backend Frameworks](https://bentoml.com/blog/breaking-up-with-flask-amp-fastapi-why-ml-model-serving-requires-a-specialized-framework)
+- [Bentoml quickstart](https://docs.bentoml.com/en/v1.1.11/quickstarts/deploy-an-iris-classification-model-with-bentoml.html)
+
+*Note that we use bentoml 1.1 so consider this when looking at the correct docomuntion pages*
+
+### Key Concepts:
+- service
+- inputs and outputs
```
Comment (giladd123) [inline 10-AI Services/10.2-bentoml.md:12]: Letters that are capital

File: 10-AI Services/10.2-bentoml.md [hunk 25]
```diff
@@ -0,0 +1,19 @@
...
+- runners
+- bento files
+- adaptive batching
+
+### Final Exercise:
+- Compare bentoml to other model deployment frameworks
+- Deploy the model you used in the ai-services chapter with bentoml and deploy it on openshift
```
Comment (giladd123) [inline 10-AI Services/10.2-bentoml.md:19]: `AI Services`

File: 11-miscellaneous/11-miscellaneous.md [hunk 26]
```diff
@@ -1,12 +1,15 @@
...
 ### Development Methodology
-##### ETC: 1 day
 - https://www.youtube.com/watch?v=-SPFHwqiBvo
 - #### DevOps
     1. https://www.youtube.com/watch?v=kpuJSpp3hhA
     2. https://www.youtube.com/watch?v=H-R2bCXfz8I
 - #### software architecture
     https://thenewstack.io/from-monolith-to-microservices/
+- #### POC products
+  - read about the following products:
```
Comment (giladd123) [inline 11-miscellaneous/11-miscellaneous.md:11]: R

File: 11-miscellaneous/11-miscellaneous.md [hunk 27]
```diff
@@ -1,12 +1,15 @@
 # Miscellaneous
 
 ### Development Methodology
-##### ETC: 1 day
 - https://www.youtube.com/watch?v=-SPFHwqiBvo
```
Comment (giladd123) [inline 11-miscellaneous/11-miscellaneous.md:4]: Name to links

File: 11-miscellaneous/11.1-reward for finishing do not open.md [hunk 28]
```diff
@@ -0,0 +1,16 @@
+# Reward
+
+Great Job, you finished the team's hafifa. Here are some memes that you probably wouldn't get.
+
+[!CAUTION]
+*If you look at them before finishing, you will have to go work without a mouse for a week.*
```
Comment (giladd123) [inline 11-miscellaneous/11.1-reward for finishing do not open.md:6]: Nice

**Review by giladd123 — CHANGES_REQUESTED** [review]
Please please please notice capital letters (even on chapters not changed) this annoys me so much I can't handle that

File: 05-jupyter/05-jupyter.md (hunk 14 above)
Comment (amoosebitmymom) [reply 05-jupyter/05-jupyter.md:21]: Yes, lots to add in this chpater:
1. Kernel
2. IPython
3. Who is Jovyan
4. Websockets

File: 07-mlops/07-mlops-concept.md
Comment (amoosebitmymom) [inline 07-mlops/07-mlops-concept.md:1]: Add to the key concepts 
1. MLops Vs DevOps

**Pull request merged**